src/mbgl/style/expression/distance.cpp

```diff
@@ -20,43 +20,46 @@
 namespace mbgl {
 namespace {
 
-const std::size_t MinPointsSize = 100;
-const std::size_t MinLinePointsSize = 50;
-const double InvalidDistance = std::numeric_limits<double>::quiet_NaN();
-const double InfiniteDistance = std::numeric_limits<double>::infinity();
-const mapbox::cheap_ruler::CheapRuler::Unit UnitInMeters = mapbox::cheap_ruler::CheapRuler::Unit::Meters;
+constexpr std::size_t MinPointsSize = 100;
+constexpr std::size_t MinLinePointsSize = 50;
+constexpr double InvalidDistance = std::numeric_limits<double>::quiet_NaN();
+constexpr double InfiniteDistance = std::numeric_limits<double>::infinity();
+constexpr mapbox::cheap_ruler::CheapRuler::Unit UnitInMeters = mapbox::cheap_ruler::CheapRuler::Unit::Meters;
 
 // Inclusive index range for multipoint or linestring container
 using IndexRange = std::pair<std::size_t, std::size_t>;
 
-std::size_t getRangeSize(const IndexRange& range) {
+std::size_t getRangeSize(const IndexRange& range) noexcept {
     return range.second - range.first + 1;
 }
 
-bool isRangeSafe(const IndexRange& range, const std::size_t threshold) {
+bool isRangeSafe(const IndexRange& range, const std::size_t threshold) noexcept {
     auto ret = (range.second >= range.first && range.second < threshold);
     if (!ret) {
         mbgl::Log::Error(mbgl::Event::Style, "Index is out of range");
     }
     return ret;
 }
 
-std::pair<std::optional<IndexRange>, std::optional<IndexRange>> splitRange(const IndexRange& range, bool isLine) {
-    if (range.first > range.second) return std::make_pair(std::nullopt, std::nullopt);
-    auto size = getRangeSize(range);
+std::pair<std::optional<IndexRange>, std::optional<IndexRange>> splitRange(const IndexRange& range,
```

Comment:
Somehow, `std::make_pair` is not `noexcept` and nor are the constructors of `std::pair`.

Reply:
Actually ended up being pretty minor